src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -493,3 +502,18 @@
       return(list(error = e))
    })
 })
+
+.rs.addFunction("preview_data_import_async", function(jsonData)
+{
+   tryCatch({
+      params = jsonlite::fromJSON(jsonData)
```

Comment:
Updated. I spent some time thinking about this and discussing with @jmcphers. My plan is to make this a bit more generic overtime and, as a start, reuse this in the profiler feature. I would also like to explore using a socket through `readRDS` and `saveRDS` to avoid leaving temp files open. For now it's just the temp file as suggested, but want to mention this in case there is feedback.